kzg/src/eip_7594.rs

```diff
@@ -414,69 +414,40 @@ pub fn recover_cells_and_kzg_proofs<
 >(
     recovered_cells: &mut [[TFr; FIELD_ELEMENTS_PER_CELL]],
     recovered_proofs: Option<&mut [TG1]>,
-    cell_indicies: &[usize],
+    cell_indices: &[usize],
     cells: &[[TFr; FIELD_ELEMENTS_PER_CELL]],
     s: &TKZGSettings,
 ) -> Result<(), String> {
-    if recovered_cells.len() != CELLS_PER_EXT_BLOB
-        || recovered_proofs
-            .as_ref()
-            .is_some_and(|it| it.len() != CELLS_PER_EXT_BLOB)
-    {
-        return Err("Invalid output array length".to_string());
-    }
+    let cells_len = cells.len();
+    let proofs_len = recovered_proofs.as_ref().map_or(0, |it| it.len());
 
-    if cells.len() != cell_indicies.len() {
-        return Err(
-            "Cell indicies mismatch - cells length must be equal to cell indicies length"
-                .to_string(),
-        );
+    if !(recovered_cells.len() == CELLS_PER_EXT_BLOB && 
+          (proofs_len == 0 || proofs_len == CELLS_PER_EXT_BLOB) && 
+          cells_len == cell_indices.len() && 
+          cells_len <= CELLS_PER_EXT_BLOB && 
+          cells_len >= CELLS_PER_EXT_BLOB / 2) {
+        return Err("Cell indices mismatch - invalid length".into());
```

Comment:
I believe these changes have very little performance improvement, with a big cost of making errors much more obscure.